sklearn/ensemble/tests/test_forest.py

```diff
@@ -1443,6 +1443,42 @@ def test_max_samples_exceptions(name, max_samples, exc_type, exc_msg):
         est.fit(X, y)
 
 
+@pytest.mark.parametrize('name', FOREST_REGRESSORS)
+def test_max_samples_boundary_regressors(name):
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_reg, y_reg, train_size=0.7, test_size=0.3, random_state=0)
+
+    ms_1_predict = FOREST_REGRESSORS[name](
+        max_samples=1.0, random_state=0).fit(
+        X_train, y_train).predict(X_test)
+    ms_None_predict = FOREST_REGRESSORS[name](
+        max_samples=None, random_state=0).fit(
+        X_train, y_train).predict(X_test)
```

Comment:
I find the following easier to read:

```suggestion
    model_ms_1 = FOREST_REGRESSORS[name](max_samples=1.0, random_state=0)
    ms_1_predict = model_ms_1.fit(X_train, y_train).predict(X_test)

    model_ms_None = FOREST_REGRESSORS[name](max_samples=None, random_state=0)
    ms_None_predict = model_ms_None.fit(X_train, y_train).predict(X_test)
```

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20159/commits/3d0005f0869205c6cae21c2d65ea5f3c815e4f82